core/src/operations/mod.rs

```diff
@@ -399,10 +399,9 @@ pub enum Operation {
     /// new root value computed based on the provided path. Everything else on the stack remains the
     /// same.
     ///
-    /// If the boolean parameter is set to false, at the end of the operation the merkle set with
-    /// the specified root will be removed from the advice provider. Otherwise, the advice
-    /// provider will keep track of both, the old and the new merkle sets.
-    MrUpdate(bool),
+    /// The tree will always be copied into a new instance, meaning the advice provider will keep
+    /// track of both the old and new Merkle sets.
+    MrUpdate,
```

Comment:
nit: I'd change the terminology a bit not to refer to "Merkle sets" both here and in the paragraph right above this one. Here, we can probably say just "Merkle tree". In the above paragraph, we could probably say "via the advice provider".

Reply:
Fixed